Sming/Components/OtaUpgrade/README.rst

```diff
@@ -0,0 +1,262 @@
+   in possession of the secret encryption key can generate upgrade files that decrypt successfully.
+   There is, however, one catch: Unlike signing, encryption *can* be broken if an attacker is able to extract the
+   decryption key (which is identical to the encryption key) from flash memory, in which case all current and future
+   file encrypted with the same key are compromised. Moreover, the attacker will be able to generate new valid upgrade 
+   files modified to his or her agenda. Hence, **you should only every rely on encryption if it is impossible for an 
+   attacker to gain physical access to your device(s)**. But otherwise, you shouldn't have stored confidential data on 
+   such device(s) in the first place!
+   Conversely, you should *not* encrypt upgrade files that do not contain confidential data, to avoid the risk of 
+   accidentally exposing a key you might want to reuse later. For this reason, encryption is disabled by default.
+   
+   Note: To mitigate a catastrophic security breach when the encryption key is revealed involuntarily, encryption and 
+   signing can be enabled at the same time. This way, an attacker (who probably has access to your WiFi by now) will at
+   least be unable to take over more devices wirelessly. But keep in mind: it is still not a good idea to store 
+   confidential data on an unsecured device!
+
+   Note also that the described weakness is not a property of the selected encryption algorithm, but a rather general 
+   one. It can only be overcome by encrypting the communication channel instead of the upgrade file, e.g. with TLS, 
+   which uses a key exchange protocol to negotiate a temporary encryption key that is never written to flash memory. But 
+   even then, it is still unwise to embed confidential data into the firmware of a device that is physically accessible
+   to an attacker - now you have been warned!
+
+.. envvar:: OTA_KEY
+
+   Path to the secret encryption/signing key. The default is ``ota.key`` in the root directory of your project. If the 
+   key file does not exist, it will be generated during the first build. It can also be (re-)generated manually using the
+   following command (usually as part of a :ref:`ota-rollover-process`):
+   
+   ::
+
+      make ota-genkey
+
+   The key file must be kept secret for obvious reasons. In particular, set up your .gitignore (or equivalent VCS 
+   mechanism) carefully to avoid accidentally pushing the key file to a public repository.
+   
+   By pointing ``OTA_KEY`` to a shared location, the same key file can be used for multiple projects, even if their 
+   security settings differ, since the key file format is independent of the security settings. (In fact, it is just a 
+   string of random numbers, from which the actual algorithm keys are derived.)
+
+.. envvar:: OTA_ENABLE_DOWNGRADE
+
+   By default, ``OtaUpgradeStrem`` refuses to downgrade to an older firmware version, in order to prevent an attacker 
```

Comment:
`OtaUpgradeStrem` should be `OtaUpgradeStream`. Run spell-checker here.

Reply:
I did that, but my spell checker was "smart" enough to ignore identifiers...